Sources/SwiftParser/Expressions.swift

```diff
@@ -1630,7 +1630,7 @@ extension Parser {
             valueExpression: value,
             trailingComma: comma,
             arena: self.arena)))
-          if key.is(RawMissingExprSyntax.self), colon.isMissing, value.is(RawMissingExprSyntax.self) {
+          if key.is(RawMissingExprSyntax.self) || colon.isMissing || value.is(RawMissingExprSyntax.self) {
```

Comment:
I am not sure if this is superior in general. IIUC if you have 
```swift
[
  1: "one",
  2: ,
  3: "three"
]
```
we wouldn’t be stopping dictionary literal parsing at the missing value for `2` and thus not add `3: "three"` to the string literal.

What do `key` and `value` contain when we synthesize the missing `colon`?

Reply:
Thanks for the pointer, I'll try to investigate further.

Interestingly, your example produces _no_ diagnostics for me with the original implementation 😅

Reply:
Oh, I forgot that my example will only start producing diagnostics once https://github.com/apple/swift-syntax/pull/739 is merged.